lms/djangoapps/email_marketing/tasks.py

```diff
@@ -120,6 +120,10 @@ def update_user(self, sailthru_vars, email, site=None, new_user=False):
                                  max_retries=email_config.sailthru_max_retries)
 
 
+def default_site(self, site):
+    return site and site.get('id') == settings.SITE_ID
```

Comment:
I think I am changing my mind here.
I think the logic here should be `return not site or site.get('id') == settings.SITE_ID`
Basically, we say, the site is default site if the site is `None`, or, the site is the same as the one in the settings.
This way, in line #191, you can just do `if not is_default_site(site):`